### DISCUSSION TITLE
[Feature request] Option to automatically delete nodes from database.

### DISCUSSION THREAD

**Discussion by ckambiselis** [meshtastic/firmware#5687] [Ideas]
Could an option be added in the settings for nodes to be deleted from the NodeDB after X amount of time since the node was last heard, and the option to keep selected nodes forever?

**Comment by fifieldt** [comment]
Nodes themselves don't have reliable time. This feature request probably needs to be sent to the android/ios apps, as it would be difficult to fix in firmware.

https://github.com/meshtastic/Meshtastic-Android
https://github.com/meshtastic/Meshtastic-Apple

**Reply by ckambiselis** [reply]
Well, the time would be based on the node that hears, and the exact time wouldn't really be that important anyway, since deleting older nodes would be more about "housekeeping" than anything else, a node that passed by a month ago, is irrelevant at this point.

Isn't the database populated even when the node is not connected to the app, or do you mean the app could automatically shift through the database and delete them when connected?

**Comment by garthvh** [comment]
You can delete nodes that you don't want and favorite ones you want to keep.

**Reply by ckambiselis** [reply]
Well the "automatically" is a big point in my request.

Where is the "favourite" marking, I don't see it on the Android app.

**Reply by bodhar** [reply]
> Where is the "favourite" marking, I don't see it on the Android app.

Looks like it's [on the way](https://github.com/meshtastic/Meshtastic-Android/pull/1520), requiring some work on the app before it'll be implemented. 

Will be well received when it arrives!
